src/app/pages/apps/components/app-detail-view/app-detail-view.component.ts

```diff
@@ -72,24 +76,30 @@ export class AppDetailViewComponent implements OnInit, AfterViewInit {
   private listenForRouteChanges(): void {
     this.activatedRoute.params
       .pipe(
-        map((params) => params.appId as string),
-        filter(Boolean),
+        filter((params) => {
+          return !!(params.appId as string) && !!(params.catalog as string) && !!(params.train as string);
+        }),
         tap(() => {
           this.isLoading$.next(true);
           this.similarAppsLoading$.next(true);
         }),
         untilDestroyed(this),
       )
-      .subscribe((appId) => {
+      .subscribe(({ appId, catalog, train }: { appId: string; catalog: string; train: string }) => {
```

Comment:
It looks like a good candidate to extract into an interface and reuse it in the ChartWizard: https://github.com/truenas/webui/pull/8080/files#diff-a8d1a9b3ab6a9e16e3aa1338554dc9c9f7f12e759e571686cf1d6df5020d13bcR142